pkg/apis/application/v1alpha1/app_types.go

```diff
@@ -76,6 +77,7 @@ spec:
                     namespace:
                       type: string
                   required: ["name", "namespace"]
+              requried: ["inCluster"]
```

Comment:
```suggestion
              required: ["inCluster"]
```

I'm not sure. Either way the default will be false due to K8s / Go. I think in cluster-operator we set it just to be explicit.

Maybe if it's true some validation that there is no secret would be useful?

Reply:
ok, so it is not required?

Reply:
Yes it is not required.

Reply:
ok, removing then, I thought we had it required, but as you explained it is just for clarity not technical requirement